NYPL-discovery/discovery-front-end
# Scc 2635

**What's this do?**
This adds a new property to the store - `resultSelection`. This will store the URL of the results page (either a search results page or a subject heading results page) and the bib id of the selected result, when a result from `ResultsList` is clicked. This data is used on the `BibPage` for the url for the 'Back to search results' link. The link will only display if the `resultSelection.bibId` matches the ID of the bib page being shown. This accounts for back button weirdness.

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SCC-2635

I did ask Sean about details here. He said, "[...] if you came from SHEP you should go back to SHEP, but I don’t know how complex that might be."

**Do these changes have automated tests?**
Necessary changes to automated tests and added tests that cover some of the implementation

**How should this be QAed?**
Check "Back to search results" link coming from search results and subject heading show page. Also, navigate to one bib, then to another, go back to first bib. The "Back to search results" link will not be there anymore, because the url in state is not necessarily relevant anymore.

**Dependencies for merging? Releasing to production?**
Merge after RC launch

**Has the application documentation been updated for these changes?**

**Did someone actually run this code to verify it works?**
I did
